### DISCUSSION TITLE
Question: how to validate thousands of tokens without send anything to them?

### DISCUSSION THREAD

**Discussion by goncalogoncalves** [kreait/firebase-php#511] [Q&A] [answered]
Hello

Is it possible to check if thousands of tokens are valid without send anything to them?

I can do

`$messaging = $this->factory->createMessaging();
$message = CloudMessage::withTarget('token', $token)
            ->withNotification([
                'title' => 'Hello',
                'body' => 'Have a nice day'
            ]);
$resultValidate = $messaging->validate($message);`

to validate one token, and I can do 

`$messaging->sendMulticast(`

to send a push to several tokens. How can I join the two? to send a "validate_only" to a multicast?

Thank you

**Comment by jeromegamez** [comment]
That's a good one! I will get to it once after I managed finishing the PHP 8 stuff 💪

**Comment by jeromegamez** [comment]
I just created #520 which hopefully satisfies your requirements 😅. Let me know what you think and if this works for you 🙏

**Comment by goncalogoncalves** [comment]
Awesome! it looks perfect to me, good job and thank you! 👏

**Comment by jeromegamez** [comment, marked as answer]
Multi token validation was released with [5.14](https://github.com/kreait/firebase-php/releases/tag/5.14.0) 🥳 . Let me know if something doesn't work 💪
